tests/proxy/replicated_map/replicated_map_test.go

```diff
@@ -0,0 +1,345 @@
+	"log"
+	"reflect"
+	"sort"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+)
+
+var rmp core.ReplicatedMap
+var client hazelcast.IHazelcastInstance
+
+func TestMain(m *testing.M) {
+	remoteController, err := NewRemoteControllerClient("localhost:9701")
+	if remoteController == nil || err != nil {
+		log.Fatal("create remote controller failed:", err)
+	}
+	cluster, err := remoteController.CreateCluster("3.9", DEFAULT_XML_CONFIG)
+	remoteController.StartMember(cluster.ID)
+	client, _ = hazelcast.NewHazelcastClient()
+	rmp, _ = client.GetReplicatedMap("myReplicatedMap")
+	m.Run()
+	rmp.Clear()
+	client.Shutdown()
+	remoteController.ShutdownCluster(cluster.ID)
+}
+
+func TestReplicatedMapProxy_Name(t *testing.T) {
+	name := "myReplicatedMap"
+	AssertEqualf(t, nil, name, rmp.Name(), "Name() failed")
+}
+
+func TestReplicatedMapProxy_ServiceName(t *testing.T) {
+	serviceName := common.SERVICE_NAME_REPLICATED_MAP
+	AssertEqualf(t, nil, serviceName, rmp.ServiceName(), "ServiceName() failed")
+}
+
+func TestReplicatedMapProxy_PartitionKey(t *testing.T) {
+	name := "myReplicatedMap"
+	AssertEqualf(t, nil, name, rmp.PartitionKey(), "PartitionKey() failed")
+}
+
+func TestReplicatedMapProxy_Destroy(t *testing.T) {
+	testKey := "testingKey"
+	testValue := "testingValue"
+	rmp.Put(testKey, testValue)
+	rmp.Destroy()
+	rmp, _ := client.GetReplicatedMap("myReplicatedMap")
+	res, err := rmp.Get(testKey)
+	AssertNilf(t, err, res, "Destroy() failed")
+}
+
+func TestReplicatedMapProxy_Put(t *testing.T) {
+	defer rmp.Clear()
+	testKey := "testingKey"
+	testValue := "testingValue"
+	newValue := "newValue"
+	_, err := rmp.Put(testKey, testValue)
+	res, err := rmp.Get(testKey)
+	AssertEqualf(t, err, res, testValue, "Put() failed")
+	oldValue, err := rmp.Put(testKey, newValue)
+	AssertEqualf(t, err, oldValue, testValue, "Put()  failed")
+}
+
+func TestReplicatedMapProxy_PutWithTtl(t *testing.T) {
+	defer rmp.Clear()
+	testKey := "testingKey"
+	testValue := "testingValue"
+	rmp.Put(testKey, testValue)
+	oldValue, err := rmp.PutWithTtl(testKey, "nextValue", 100, time.Second)
+	AssertEqualf(t, err, oldValue, testValue, "PutWithTtl()  failed")
+	res, err := rmp.Get(testKey)
+	AssertEqualf(t, err, res, "nextValue", "PutWithTtl() failed")
+}
+
+func TestMapProxy_PutWithTtlWhenExpire(t *testing.T) {
+	defer rmp.Clear()
+	testKey := "testingKey"
+	testValue := "testingValue"
+	rmp.Put(testKey, testValue)
+	rmp.PutWithTtl(testKey, "nextValue", 1, time.Millisecond)
+	time.Sleep(2 * time.Second)
+	res, err := rmp.Get(testKey)
+	AssertNilf(t, err, res, "PutWithTtl() failed")
+}
+
+func TestReplicatedMapProxy_PutAll(t *testing.T) {
+	defer rmp.Clear()
+	testMap := make(map[interface{}]interface{})
+	for i := 0; i < 10; i++ {
+		testMap["testingKey"+strconv.Itoa(i)] = "testingValue" + strconv.Itoa(i)
+	}
+	err := rmp.PutAll(testMap)
+	AssertErrorNil(t, err)
+	entryList, err := rmp.EntrySet()
+	AssertErrorNil(t, err)
+	for _, pair := range entryList {
```

Comment:
if `entryList` did not include all the required elements, this test would still pass.
It will be better if you write a utility method that will check equality of two slices and use that. Another utility method may also be necessary for getting the slice from map.